book/src/exec.md

```diff
@@ -0,0 +1,174 @@
+- **have the authority** to spend and create the notes they receive and produce the proofs required to perform the action,
+- **know** the content of intentVPs of the users (necessary to be able to satisfy them),
+- **don't know** the identities of the users.
+
+When solvers receive partial solutions from the gossip network, they must check all of the proofs attached to them.
+
+#### Local `cm` trees
+
+To store note commitments of the created notes, a local commitment tree `CMtree` is created. After the transaction is finalized, the tree will be published on the blockchain along with the transaction.
+
+### Partial vs final match
+
+After the solver matches the intents, two cases are possible:
+1. At least one of the total per-token balances computed by summing up the per-token balances of partial transactions doesn't equal the balancing value. In this case, the solver sends the data to the gossip network.
+2. All total per-token balances are equal to the balancing values. A valid transaction can be created and published.
+
+**Note**: in the current implementation we assume a simpler model with no partial solving.
+
+**Note**: solvers don't need to be identified as all actions are authorized by user/app VPs. However, if they want to receive fees, they need to have an address on the chain.
+
+## Step 4: Finalize
+
+After the intents are matched with satisfaction of all involved parties, the transaction is published on the blockchain. The local CMTree and all of the proofs created are published on the blockchain.
+
+## Examples
+
+Let's see how different use cases can be handled with this execution model.
+
+**Note**: For simplicity in the examples below we assume that the balancing value `v` is equal to zero.
+
+### 1. Three-party barter
+
+Three parties, Alice, Bob, and Charlie, are looking for some assets in exchange for something else. Their intents can be matched into a three-party bartering cycle. Let's see step by step how this happens.
+
+**Note**: parties don't require three-party bartering explicitly.
+
+![img.png](img/exec_3_party.png)
+
+**Step 1-2**: The users define their intents and create intentVPs. It doesn't have to happen at the same time for all users, but for simplicity we describe it as one step. The intentVPs of all three users have the same structure: users are willing to spend their asset (Alice - a star, Bob - a dolphin, Charlie - a tree) in exchange for some other asset. Once the user receives the desired asset, the intent token note is spent. In addition to that, all three users also create their initial partial transactions spending the asset they are ready to give away (Alice ptx, Bob ptx, and Charlie ptx).
+
+**Step 3**: A solver sees Alice's ptx and Bob's ptx, matches them together, and creates a new partial transaction. Alice's intentVP is satisfied, and her intent note is spent. 
```

Comment:
I am thinking about the general case where a solver receives `n` partial transactions. Given two ptx, the solver can always do this step 3, and so it creates o(n²) ptx. This seems to be a lot, but maybe I didn't get how it works in practice...

Reply:
If a solver sees a ptx sending BTC and expecting NAM and another ptx sending XAN and expecting ETC, it doesn't make sense for the solver to match them

Plus, matching that many ptx together in all possible combinations considering that no more than n/2 of them will be a part of real transactions seems like a waste of time for the solver

Reply:
Yes, that's my point. With this description of solver, it seems that they are waiting their time. From what you said, in practice their have other conditions that refine their search. Thanks!